src/sentry/snuba/discover.py

```diff
@@ -770,82 +770,107 @@ def get_performance_facets(
-            # TODO(Kevan): Check using having vs where before mainlining
-            having=[excluded_tags],
+            orderby=["-count"],
             dataset=Dataset.Discover,
             limit=limit,
-            referrer=referrer,
-            turbo=sample,
+            referrer="{}.{}".format(referrer, "all_transactions"),
         )
-        top_tags = [r["tags_key"] for r in key_names["data"]]
-        if not top_tags:
+        counts = [r["count"] for r in key_names["data"]]
+        if len(counts) != 1 or counts[0] == 0:
             return []
 
     results = []
     snuba_filter.conditions.append([aggregate_column, "IS NOT NULL", None])
 
-    # Only enable sampling if over 10000 values
-    sampling_enabled = key_names["data"][0]["count"] > 10000
-    options_sample_rate = options.get("discover2.tags_performance_facet_sample_rate") or 0.1
+    # Aggregate for transaction
+    transaction_aggregate = key_names["data"][0]["aggregate"]
 
-    sample_rate = options_sample_rate if sampling_enabled else None
+    # Dynamically sample so at least 10000 transactions are selected
+    transaction_count = key_names["data"][0]["count"]
+    sampling_enabled = transaction_count > 10000
+    # Log growth starting at 10,000
+    target_sample = 10000 * (math.log(transaction_count, 10) - 3)
 
-    max_aggregate_tags = 20
-    aggregate_tags = []
-    for i, tag in enumerate(top_tags):
-        if i >= len(top_tags) - max_aggregate_tags:
-            aggregate_tags.append(tag)
+    dynamic_sample_rate = 0 if transaction_count <= 0 else (target_sample / transaction_count)
+    options_sample_rate = (
+        options.get("discover2.tags_performance_facet_sample_rate") or dynamic_sample_rate
+    )
 
-    if orderby is None:
-        orderby = []
+    sample_rate = options_sample_rate if sampling_enabled else None
 
-    if aggregate_tags:
-        with sentry_sdk.start_span(op="discover.discover", description="facets.aggregate_tags"):
-            conditions = snuba_filter.conditions
-            conditions.append(["tags_key", "IN", aggregate_tags])
-            tag_values = raw_query(
-                aggregations=[
-                    [aggregate_function, aggregate_column, "aggregate"],
-                    ["count", None, "count"],
-                ],
-                conditions=conditions,
-                start=snuba_filter.start,
-                end=snuba_filter.end,
-                filter_keys=snuba_filter.filter_keys,
-                orderby=orderby + ["tags_key"],
-                groupby=["tags_key", "tags_value"],
-                dataset=Dataset.Discover,
-                referrer=referrer,
-                sample=sample_rate,
-                turbo=sample_rate is not None,
-                limitby=[TOP_VALUES_DEFAULT_LIMIT, "tags_key"],
-            )
-            results.extend(
+    excluded_tags = [
+        "tags_key",
+        "NOT IN",
+        ["trace", "trace.ctx", "trace.span", "project", "browser", "celery_task_id"],
```

Comment:
Just noticed but why is `browser` in this list too? I don't mind excluding it, just curious about the reasoning.

Reply:
I had removed it since `browser` is really variable, `browser.name` is probably sufficient. I might bring it back after I see with prod data.